lighthouse-core/gather/gatherers/js-usage.js

```diff
@@ -104,6 +111,26 @@ class JsUsage extends FRGatherer {
       usageByUrl[url] = scripts;
     }
 
+    // Usages alone do not always generate an exhaustive list of scripts in timespan and snapshot.
+    // For audits which use this for url/scriptId mappings, we can include an empty usage object.
+    // TODO(FR-COMPAT): Enable this logic for legacy and navigation or make a separate artifact for url/scriptId mappings.
+    if (context.gatherMode !== 'navigation') {
+      for (const scriptParsedEvent of this._scriptParsedEvents) {
```

Comment:
want to wrap this up in a separate method? `_addMissingParsedScriptsToUsage` or something?